checks/__init__.py

```diff
@@ -564,6 +586,14 @@ def run(self):
                     tb=traceback.format_exc()
                 )
             instance_statuses.append(instance_status)
+            # Generate empty metadata if nonexistent
+            service_metadata_count = len(self.service_metadata)
+            if service_metadata_count > i + 1:
```

Comment:
`self.service_metadata` is indexing integration metadata dictionnary by instance. 
This mechanism makes sure that the number of metadata dictionnaries generated matches the number of instances: if an instance does not produce any meta (purposely or because it's failing), we complete `self.service_metadata` adding some padding.